front/main/app/models/infobox-builder.js

```
@@ -126,8 +136,41 @@ const InfoboxBuilderModel = Ember.Object.extend({
 	},
 
 	/**
+	 * @desc sets a new value of the default field
+	 * on the given title element
+	 *
+	 * @param {Object} item
+	 * @param {Boolean} value
+	 * @returns {void}
+	 */
+	editTitleItem(item, value) {
+		const index = this.get('infoboxState').indexOf(item),
+			defaultValue = value ? '{{PAGENAME}}' : '';
+
+		this.set(`infoboxState.${index}.data.default`, defaultValue);
+	},
+
+	/**
+	 * @desc sets a new value of the label field
+	 * on the given row (data) element
+	 *
+	 * @param {RowItem} item
+	 * @param {string} value
+	 * @returns {void}
+	 */
+	editRowItem(item, value) {
+		const index = this.get('infoboxState').indexOf(item);
+
+		this.set(`infoboxState.${index}.data.label`, value);
+
+		if (value.trim().length) {
```

no, I could make it bullet proof and add the check but on other hand I'd probably prefer to throw error rather than silently skip setting source value